azurerm/resource_arm_cosmos_table.go

```diff
@@ -0,0 +1,153 @@
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.CosmosAccountName,
+			},
+		},
+	}
+}
+
+func resourceArmCosmosTableCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).cosmosAccountsClient
+	ctx := meta.(*ArmClient).StopContext
+
+	name := d.Get("name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+	account := d.Get("account_name").(string)
+
+	if requireResourcesToBeImported && d.IsNewResource() {
+		existing, err := client.GetTable(ctx, resourceGroup, account, name)
+		if err != nil {
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("Error checking for presence of creating Cosmos Table %s (Account %s): %+v", name, account, err)
+			}
+		} else {
+			id, err := azure.CosmosGetIDFromResponse(existing.Response)
+			if err != nil {
+				return fmt.Errorf("Error generating import ID for  Cosmos Table '%s' (Account %s)", name, account)
+			}
+
+			return tf.ImportAsExistsError("azurerm_cosmos_mongo_database", id)
+		}
+	}
+
+	db := documentdb.TableCreateUpdateParameters{
+		TableCreateUpdateProperties: &documentdb.TableCreateUpdateProperties{
+			Resource: &documentdb.TableResource{
+				ID: &name,
+			},
+			Options: map[string]*string{},
+		},
+	}
+
+	future, err := client.CreateUpdateTable(ctx, resourceGroup, account, name, db)
+	if err != nil {
+		return fmt.Errorf("Error issuing create/update request for Cosmos Table %s (Account %s): %+v", name, account, err)
+	}
+
+	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("Error waiting on create/update future for Cosmos Table %s (Account %s): %+v", name, account, err)
+	}
+
+	resp, err := client.GetTable(ctx, resourceGroup, account, name)
+	if err != nil {
+		return fmt.Errorf("Error making get request for Cosmos Table %s (Account %s): %+v", name, account, err)
+	}
+
+	id, err := azure.CosmosGetIDFromResponse(resp.Response)
+	if err != nil {
+		return fmt.Errorf("Error creating Cosmos Table '%s' (Account %s) ID: %v", name, account, err)
+	}
+	d.SetId(id)
+
+	return resourceArmCosmosTableRead(d, meta)
+}
+
+func resourceArmCosmosTableRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).cosmosAccountsClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := azure.ParseCosmosTableID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.GetTable(ctx, id.ResourceGroup, id.Account, id.Table)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[INFO] Error reading Cosmos Table %s (Account %s) - removing from state", id.Table, id.Account)
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("Error reading Cosmos Table %s (Account %s): %+v", id.Table, id.Account, err)
+	}
+
+	if props := resp.TableProperties; props != nil {
+		d.Set("name", props.ID)
+		d.Set("resource_group_name", id.ResourceGroup)
+		d.Set("account_name", id.Account)
```

Comment:
(as above) we can move these outside of this function since they're not dependent on `props`